src/components/MapView/Legends/index.tsx

```diff
@@ -76,35 +89,113 @@ function Legends({ classes, layers }: LegendsProps) {
 }
 
 // Children here is legendText
-function LegendItem({ classes, title, legend, children }: LegendItemProps) {
+function LegendItem({
+  classes,
+  id,
+  title,
+  legend,
+  type,
+  opacity: initialOpacity,
+  children,
+}: LegendItemProps) {
+  const map = useSelector(mapSelector);
+  const [opacity, setOpacityValue] = useState<number | number[]>(
+    initialOpacity || 0,
+  );
+
+  const [toggle, setToggle] = useState(true);
+  const toggleChecked = (prev: Boolean) => {
+    setToggle(!prev);
+  };
+
+  const handleChangeOpacity = (event: object, newValue: number | number[]) => {
+    setOpacityValue(newValue);
+  };
+
+  const layerTypes = ['wms', 'nso', 'point_data', 'impact'];
+  const [layerId, opacityType] = (e => {
+    switch (e) {
+      case 'wms':
+        return [`layer-${id}`, 'raster-opacity'];
+      case 'impact':
+      case 'nso':
+        return [`layer-${id}-fill`, 'fill-opacity'];
+      case 'point_data':
+        return [`layer-${id}-circle`, 'circle-opacity'];
+      default:
+        return ['', ''];
+    }
+  })(type);
+
+  const allLayers = map!.getStyle().layers;
+  const isAvailable = allLayers!.find(layer => layer.id === layerId);
```

Comment:
You should use `some` instead so you don't end up casting everywhere. Also, maybe `isLayerActive` is a better name here?